rfc/rfc-0000-k8s-events.md

```diff
@@ -0,0 +1,162 @@
+definition to the cluster) are two very common events, with associated causes, that currently need further exposure to
+users.
+
+# What it is
+
+[what-it-is]: #what-it-is
+
+## Developer View
+
+Cartographer presents useful information to developers in the **owner** (workload/deliverable). This information is
+derived from stamped objects and contributing resources such as **templates** and **blueprints**:
+
+This RFC addresses the absence of awareness that "something is happening", especially after a developer has committed
+new code. With event's we can provide information that show's activity and causality not present in the current snapshot
+of state.
+
+We add events to the **Owners** (Workload/Deliverable) to enable debug and reasoning about Cartographer's behavior.
+
+## Operator View
+
+The situation is worse for users managing the **blueprints** (Devops/Ops). Some metrics can be recorded in **
+blueprints**
+and this might be a valid topic for another RFC, however events also present a useful way for authors to avoid trawling
+through logs to get a quick overview of temporal behaviour of **owners** (workload/delivery) as they rely on the
+author's
+**blueprint**.
+
+# How it Works
+
+[how-it-works]: #how-it-works
+
+To avoid over-communicating, and making the view of event's overwhelming and dilute for user's we will start with two
+key guiding principles:
+
+1. > Let it be something that isn't satisfactorily presented in other user views (eg: `k get workload`).
+2. > Let the event list be as meaningful as is necessary, and no more.
+
+These are examples of types of events that could be emitted. This RFC makes no recommendation as to which events are
+required. The cartographer controller could emit the following events:
+
+| Reason | Message Format | Description | involvedObject |
+| --- | --- | --- | --- |
+| StampedObjectExternalSpecChange | an external actor changed the spec of `<resource>.<group>/<name>` | Our cache of the spec, and the spec we just generated match, but the API server has a different one. A lot of these is evidence of thrashing with external resources | Owner |
+| StampedObjectExternalSpecChange | an external actor changed the spec of resource: `<resource name>` for object `<resource>.<group>` | Our cache of the spec, and the spec we just generated match, but the API server has a different one. A lot of these is evidence of thrashing with external resources | Blueprint |
+| StampedObjectGetError | `<resource name>` could not retrieve `<resource>.<group>/<name>` due to error: `<error message>` | Loading a resource is failing due to a client.get issue (missing is not an error) | Owner |
+| StampedObjectObjectGetError | `<resource name>` could not retrieve `<resource>.<group>` due to error: `<error message>` | Loading a resources is failing due to a client.get issue (missing is not an error) - this could be spammy | Blueprint |
```

Comment:
`StampedObjectGetError` vs `StampedObjectObjectGetError` Is this a typo? Are these meant to be the same event, just associated with different involvedObjects?

Reply:
definitely a typo